app/src/androidTest/java/org/simple/clinic/facility/FacilityRepositoryAndroidTest.kt

```diff
@@ -31,24 +33,24 @@ class FacilityRepositoryAndroidTest {
   lateinit var facilityDao: Facility.RoomDao
 
   @Inject
-  lateinit var mappingDao: LoggedInUserFacilityMapping.RoomDao
+  lateinit var facilityMappingDao: LoggedInUserFacilityMapping.RoomDao
 
   @get:Rule
-  val rxErrorsRule = RxErrorsRule()
+  val rule = RuleChain
+      .outerRule(LocalAuthenticationRule())
+      .around(RxErrorsRule())
 
-  private lateinit var user: User
+  private val user: User
+    get() = testData.qaUser()
 
   @Before
   fun setup() {
     TestClinicApp.appComponent().inject(this)
-    database.clearAllTables()
-
-    user = testData.loggedInUser()
-    database.userDao().createOrUpdate(user)
   }
 
   @Test
   fun facilities_should_be_ordered_alphabetically() {
+    clearFacilities()
```

Comment:
This is confusing because, why are we clearing facilities in certain tests and not the others? This gives me a sense that there could be state involved in other tests. Is that the case?

Reply:
The problem was that in the older tests, the `AuthenticationRule` was clearing all data and the tests were depending on this behaviour. However, when they were broken out, this behaviour was changed and tests started breaking.

I added this clearing of data to temporarily stop the tests from breaking and to reintroduce the behaviour of having a clean slate between each test. I'll take a look at this again and see if I can remove this in a nice way.

Reply:
"Unresolving" this conversation until I can manage to take a look at what I can do to improve it.

Reply:
`LocalAuthenticationRule` was inserting facilities on its own, while at the same time, this test class was testing the behaviour of the same system. I changed this test class to not use the rule and create a user locally, and everything is fine now.